tests/models/test_taskinstance.py

```diff
@@ -1528,7 +1534,12 @@ def do_something_else(i):
             dep_context=DepContext(flag_upstream_failed=flag_upstream_failed),
             session=session,
         )
+        if is_db_isolation_mode():
+            TracebackSessionForTests.set_allow_db_access(session, True)
         completed = all(dep.passed for dep in dep_results)
+        if is_db_isolation_mode():
+            TracebackSessionForTests.set_allow_db_access(session, False)
```

Comment:
You can save the `if` as the TraceBackSessionForTest checks this internally.
```suggestion
        TracebackSessionForTests.set_allow_db_access(session, True)
        completed = all(dep.passed for dep in dep_results)
        TracebackSessionForTests.set_allow_db_access(session, False)
```

Reply:
Oh. yeah . I was not sure if this method is in main